src/realm.js

```diff
@@ -274,6 +276,9 @@ export class Realm {
   }
 
   pushContext(context: ExecutionContext): void {
+    if (this.maxStackDepth && this.contextStack.length >= this.maxStackDepth) {
```

Comment:
The same, if this.maxStackDepth is never undefined, you do not need the first null check.